### PR TITLE
HBASE-28901 checkcompatibility.py can run maven commands with parallelism

### PR DESCRIPTION
**Pull request by ndimiduk**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 40s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  9s |  |  Maven dependency ordering for branch  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  5s |  |  Maven dependency ordering for patch  |
|||| _ Other Tests _ |
|  |   |   1m 42s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6343/1/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6343 |
| Optional Tests |  |
| uname | Linux 65fbdaf390bc 5.4.0-192-generic #212-Ubuntu SMP Fri Jul 5 09:47:39 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / fdf2d210cb3b4f0e68c3be45c9ede50211b36ede |
| Max. process+thread count | 38 (vs. ulimit of 30000) |
| modules | C:  U:  |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6343/1/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 42s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  9s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  spotless  |   0m 43s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  5s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  pylint  |   0m  8s |  |  No new issues.  |
| +1 :green_heart: |  spotless  |   0m 41s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +0 :ok: |  asflicense  |   0m  0s |  |  ASF License check generated no output?  |
|  |   |   3m 22s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6343/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6343 |
| Optional Tests | dupname asflicense codespell detsecrets pylint spotless |
| uname | Linux d000ef5a4662 5.4.0-195-generic #215-Ubuntu SMP Fri Aug 2 18:28:05 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / fdf2d210cb3b4f0e68c3be45c9ede50211b36ede |
| Max. process+thread count | 46 (vs. ulimit of 30000) |
| modules | C:  U:  |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6343/1/console |
| versions | git=2.34.1 maven=3.9.8 pylint=2.15.5 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

File: dev-support/checkcompatibility.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def build_tree(java_path, verbose):
     # special hack for comparing with rel/2.0.0, see HBASE-26063 for more details
     subprocess.check_call(["sed", "-i", "2148s/3.0.0/3.0.4/g", "pom.xml"], cwd=java_path)
     mvn_cmd = ["mvn", "--batch-mode", "-DskipTests",
-               "-Dmaven.javadoc.skip=true", "package"]
+               "-Dmaven.javadoc.skip=true", "--threads=1.0C", "package"]
```
Comment (Apache9) [inline dev-support/checkcompatibility.py:144]: Just use `-T1C`?

Comment (ndimiduk) [reply dev-support/checkcompatibility.py:144]: All the other arguments are fully written out, so I followed suit.

**Review by Apache9 — APPROVED** [review]

**Pull request merged**